tools/fuzzer.cpp

```diff
@@ -0,0 +1,211 @@
+                                "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  size_t length = randomGenerator.mt64() % 1000;
+  std::string s;
+  s.reserve(length);
+  while (length--) {
+    s += availableChars[randomGenerator.mt64() % (sizeof(availableChars) - 1)];
+  }
+  builder.add(Value(s));
+}
+
+static void generateVelocypack(Builder& builder, size_t depth, RandomGenerator& randomGenerator) {
+  RandomBuilderAdditions randomBuilderAdds;
+  while (true) {
+    randomBuilderAdds = static_cast<RandomBuilderAdditions>(randomGenerator.mt64() % 8);
+    if (depth > 10 && randomBuilderAdds < 2) {
+      continue;
+    }
+    switch (randomBuilderAdds) {
+      case ADD_ARRAY: {
+        builder.openArray(randomGenerator.mt64() % 2 ? true : false);
+        size_t numMembers = randomGenerator.mt64() % 10;
+        for (size_t i = 0; i < numMembers; ++i) {
+          generateVelocypack(builder, depth + 1, randomGenerator);
+        }
+        builder.close();
+        break;
+      }
+      case ADD_OBJECT: {
+        builder.openObject(randomGenerator.mt64() % 2 ? true : false);
+        size_t numMembers = randomGenerator.mt64() % 10;
+        for (size_t i = 0; i < numMembers; ++i) {
+          std::string key = "test" + std::to_string(i);
+          builder.add(Value(key));
+          generateVelocypack(builder, depth + 1, randomGenerator);
+        }
+        builder.close();
+        break;
+      }
+      case ADD_BOOLEAN:
+        builder.add(Value(randomGenerator.mt64() % 2 ? true : false));
+        break;
+      case ADD_STRING:
+        addString(builder, randomGenerator);
+        break;
+      case ADD_NULL:
+        builder.add(Value(ValueType::Null));
+        break;
+      case ADD_UINT64:
+        builder.add(Value(randomGenerator.mt64()));
+        break;
+      case ADD_INT64: {
+        uint64_t uintValue = randomGenerator.mt64();
+        int64_t intValue;
+        memcpy(&intValue, &uintValue, sizeof(uintValue));
+        builder.add(Value(intValue));
+        break;
+      }
+      case ADD_DOUBLE: {
+        uint64_t uintValue = randomGenerator.mt64();
+        double doubleValue;
+        memcpy(&doubleValue, &uintValue, sizeof(uintValue));
+        builder.add(Value(doubleValue));
+      }
+      default:
+        break;
+    }
+    break;
+  }
+}
+
+int main(int argc, char* argv[]) {
+  VELOCYPACK_GLOBAL_EXCEPTION_TRY
+    std::vector<std::thread> threads;
+    bool isTypeAssigned = false;
+    size_t numIterations = 1;
+    size_t numThreads = 1;
+    Format format;
+
+    int i = 1;
+    while (i < argc) {
+      char const *p = argv[i];
+      if (isOption(p, "--help")) {
+        usage(argv);
+        return EXIT_SUCCESS;
+      } else if (isOption(p, "--vpack") && !isTypeAssigned) {
+        isTypeAssigned = true;
+        format = Format::VPACK;
+      } else if (isOption(p, "--json") && !isTypeAssigned) {
+        isTypeAssigned = true;
+        format = Format::JSON;
+      } else if (int value = atoi(p); value >= 1) {
```

Comment:
Let's not use `atoi` for C++. The alternative since C++17 is https://en.cppreference.com/w/cpp/utility/from_chars.